api/v1alpha1/openapi.yaml

```diff
@@ -5348,29 +5730,68 @@ components:
     AuthConfig:
       type: object
       properties:
-        authType:
+        providers:
+          type: array
+          description: List of all available authentication providers.
+          items:
+            $ref: '#/components/schemas/AuthProviderInfo'
+        defaultProvider:
           type: string
-          description: Auth type.
-        authURL:
+          description: Name of the default authentication provider.
+        organizationsEnabled:
+          type: boolean
+          description: Whether organizations are enabled for authentication.
+    AuthProviderInfo:
+      type: object
+      properties:
+        name:
           type: string
-          description: Auth URL.
-        authOrganizationsConfig:
-          $ref: '#/components/schemas/AuthOrganizationsConfig'
+          description: Unique identifier for the provider.
+        displayName:
+          type: string
+          description: Human-readable display name.
+        type:
+          type: string
+          enum: [oidc, k8s, aap, oauth2]
+          description: Type of authentication provider.
+        issuer:
+          type: string
+          description: OIDC issuer URL (for OIDC providers).
+        clientId:
+          type: string
+          description: Default client ID for OIDC/OAuth2 providers.
+        authUrl:
+          type: string
+          description: Authentication URL for the provider.
+        tokenUrl:
+          type: string
+          description: Token endpoint URL (for OAuth2 providers).
+        userinfoUrl:
+          type: string
+          description: Userinfo endpoint URL (for OAuth2 providers).
+        scopes:
+          type: array
+          items:
+            type: string
+          description: OAuth2 scopes (for OAuth2 providers).
+        usernameClaim:
+          type: array
+          items:
+            type: string
+          description: JSON path to the username claim (for OIDC/OAuth2 providers) as an array of path segments.
+        isDefault:
+          type: boolean
+          description: Whether this is the default provider.
+          default: false
```

Comment:
Duplicate with `defaultProvider` in AuthConfig? Would allow setting multiple defaults?

Reply:
removed isDefault